torchvision/ops/deform_conv.py

```diff
@@ -0,0 +1,128 @@
+import math
+
+import torch
+from torch import nn, Tensor
+from torch.nn import init
+from torch.nn.parameter import Parameter
+from torch.nn.modules.utils import _pair
+from torch.jit.annotations import Optional, Tuple
+
+
+def deform_conv2d(input, weight, offset, bias=None, stride=(1, 1), padding=(0, 0), dilation=(1, 1)):
+    # type: (Tensor, Tensor, Tensor, Optional[Tensor], Tuple[int, int], Tuple[int, int], Tuple[int, int]) -> Tensor
+    """
+    Performs Deformable Convolution, described in Deformable Convolutional Networks
+
+    Arguments:
+        input (Tensor[batch_size, in_channels, in_height, in_width]): input tensor
+        weight (Tensor[out_channels, in_channels // groups, kernel_height, kernel_width]):
+            convolution weights, split into groups of size (in_channels // groups)
+        offset (Tensor[batch_size, 2 * offset_groups * kernel_height * kernel_width,
+            out_height, out_width]): offsets to be applied for each position in the
+            convolution kernel.
+        bias (Tensor[out_channels]): optional bias of shape (out_channels,). Default: None
+        stride (int or Tuple[int, int]): distance between convolution centers. Default: 1
+        padding (int or Tuple[int, int]): height/width of padding of zeroes around
+            each image. Default: 0
+        dilation (int or Tuple[int, int]): the spacing between kernel elements. Default: 1
+
+    Returns:
+        output (Tensor[batch_sz, out_channels, out_h, out_w]): result of convolution
+    """
+
+    out_channels = weight.shape[0]
+    if bias is None:
+        bias = torch.zeros(out_channels, device=input.device, dtype=input.dtype)
```

Comment:
We can leave this as is, but ideally if the `bias` is None we would want to avoid adding it in the code.

One easy way of doing it would be something like
```python
out = # output of deform_conv2d
if bias is not None:
    out = out + bias[:, None, :, :]
```
this is slightly less efficient than if we had it in the kernel, but saves some compute if the bias doesn't exist.
Also note that we can handle this on the C++ level with `c10::optional`.

Reply:
I guess we can do this as a follow-up.
Btw, what is the best way to handle optionals and autograd::Variable?

Reply:
> I guess we can do this as a follow-up.

Sure, this is not a blocker for this PR

> Btw, what is the best way to handle optionals and autograd::Variable?

I would have expected that `c10::optional` would be enough, but if that's not the case I'd need to look a bit more. Let's let this in a follow-up